### ISSUE TITLE
Link to the profile view

### ISSUE THREAD

**Issue by alexandraciobica** [joomla/joomla-cms#20958]
I think in com_users there should be a link to the profile view from the edit profile view. I want to add it but I don't know if there is any reason why the link shouldn't be there. Can anyone help, please? 

(I am working at Enhance Users GSoC project)

**Comment by brianteeman** [comment]
I am not sure I understand what you mean. Are you talking about front end or admin.

**Comment by alexandraciobica** [comment]
Front end. 
I want to add the link here:
<img width="793" alt="screen shot 2018-07-03 at 10 37 30" src="https://user-images.githubusercontent.com/18381652/42205438-2fead5e0-7ead-11e8-8430-b1f9c328cadb.png">
To this view:
<img width="847" alt="screen shot 2018-07-03 at 10 37 40" src="https://user-images.githubusercontent.com/18381652/42205437-2fba3926-7ead-11e8-9358-33c3b522eded.png">

You have the possibility to go from profile to edit profile, but you cannot do it the other way around.

**Comment by brianteeman** [comment]
Why would you want to? What is the use case for wanting this?
If you look at any "edit" page we dont have this ability

**Comment by alexandraciobica** [comment]
You don't have it, but in Articles for example, if you edit an article and click "Save" or "Cancel" it takes you to the article, in profile view these actions take you to the same edit profile view. So in order to be able to get to the profile view, you have to add a menu item.

**Comment by brianteeman** [comment]
ah - now i understand you. I think i have seen a bug report about this in the past requesting that the save button redirected to a view of the profile - just like editing contacts or articles

The problem is that while very many sites need a user to be able to edit their profile - for example to change their password or to agree to some new terms and conditions very few sites have a need to display a profile.

So in conclusion if a site wants to display a user profile then they would have created a menu item for it.

I guess what you could do is see if it is possible to detect **if** there is a published profile menu item and then redirect to that when you save but I don't see the benefit of a general redirection myself

**Comment by SharkyKZ** [comment]
Duplicate #20485.

This is a routing issue. It's one view with different layouts but layouts aren't taken into account when routing. With old routing this could have been fixed in `UsersHelperRoute::getProfileRoute()` but it's deprecated and not used now. Not sure how this works with new routing. @Hackwar?

**Comment by alexandraciobica** [comment]
@SharkyKZ, I know there is a routing issue, I am currently trying to fix the router. I managed to make it see the layouts, but the link doesn't look how it should. 

For example, I have a menu item for "Password reset" that when you are logged in redirects to the edit profile. The link that is displayed is `index.php/password-reset?view=profile&layout=edit` where password-reset is the menu item. I think there is something that I don't understand. 
Can anyone help me, please? Or is there any good documentation for the new router? (every piece of documentation I found gives as example the com_content router and I think this one is a bit different)

**Comment by ghost** [comment]
@Hackwar can you please help @alexandraciobica?

**Comment by ghost** [comment]
closed as having Pull Request #21187
